compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -494,12 +494,17 @@ public void visit(BLangClassDefinition classDefinition) {
 
         for (BAttachedFunction func : ((BObjectTypeSymbol) classDefinition.symbol).referencedFunctions) {
             BInvokableSymbol funcSymbol = func.symbol;
+            if (Symbols.isService(classDefinition.symbol)
```

Comment:
Same comment as above. Since resources are only allowed in services do we need to check this here?